1-common-tasks/algorithms/accumulate-values.cpp

```diff
@@ -0,0 +1,49 @@
+// Accumulate values from a range
+// C++11
+
+#include <deque>
+#include <forward_list>
+#include <numeric>
+#include <vector>
+
+template <typename Collection>
+typename Collection::value_type
+sum_numbers(Collection const& collection)
+{
+    using ReturnType = typename Collection::value_type;
+    return std::accumulate(std::begin(collection), std::end(collection), static_cast<ReturnType>(0));
```

Comment:
Just curious, why use `std::begin` and `std::end` instead of `collection.begin()` and `collection.end()`?